### ISSUE TITLE
Full LogQL not yet supported?

### ISSUE THREAD

**Issue by tomgs** [grafana/loki#985]
If I'm looking at `https://github.com/grafana/loki/blob/master/docs/querying.md#range-vector-aggregation`, it seems that there's already an option for queried of the form `count_over_time({level="ERROR"}[5m])` to be evaluated properly.

However, I'm running Grafana `6.3.2` and I keep getting this message:

![image](https://user-images.githubusercontent.com/8835035/64528837-1cf1ee00-d312-11e9-9c63-e6e9fe95c194.png)

If I'm not mistaken, this is because the option for range vector aggregation was only added to `master` 5 days ago (in https://github.com/grafana/loki/pull/654), and the latest version is `6.3.5` that was release 7 days ago, right?

Or is there capability to do the above query in `6.3.5` already?

**Comment by slim-bean** [comment]
The newest loql capabilities are still experimental and are won't be available in Grafana until 6.4

We hope to have logcli updated soon as soon as we finish some API details

Please note, the current API is going to change!

**Comment by tomgs** [comment]
OK, will wait for `6.4`. Thanks!

**Comment by Horkyze** [comment]
I am experiencing the same issue on Grafana 6.4.2 and Loki from[ docker hub](https://hub.docker.com/layers/grafana/loki/latest/images/sha256-469bed72813d14737e150b3e9a76dcdd9e705308c947db889233c360769d890f)

![image](https://user-images.githubusercontent.com/1040383/66564277-ed781080-eb5f-11e9-9c8d-a1d2f32bcdad.png)

What is the current status of this issue? Should it be working?

**Comment by cyriltovena** [comment]
No this is not available yet in Grafana, @davkal do we have an issue on Grafana side ?

Also I notice the path Grafana is using currently is wrong it should be https://github.com/grafana/loki/blob/master/docs/api.md#get-lokiapiv1query_range

If you still want to use this you can @Horkyze , you can use Loki as a Prometheus datasource, although you won't have label and label value completion it will still works in dashboard and explore.

When adding Loki as a Prometheus datasource use this URL path: `http://loki-dns:port/loki`

here is some example

## Using Explore

![image](https://user-images.githubusercontent.com/1053421/66577672-cd3c5780-eb47-11e9-98a9-a31eba0eff68.png)

## Using a dashboard

![image](https://user-images.githubusercontent.com/1053421/66577917-2a380d80-eb48-11e9-97b9-ead1977be67e.png)


Basically Loki is 100% compatible with the Prometheus API for those queries :)

**Comment by Horkyze** [comment]
Thanks for quick reply. 

Could you provide you datasource config please?
Because I cant get it to work.. :/

`curl https://loki.<oursite>.net/loki/api/v1/query?query=1%2B1&time=1570721681.814`
return 404 from our proxy..

**Comment by cyriltovena** [comment]
I'm guessing you're on Loki 0.3, you need at least this commit https://github.com/grafana/loki/commit/364b5bc4f654230eb0640b89fba99be98184a610 which will be in the soon to be released 0.4

**Comment by cyriltovena** [comment]
https://hub.docker.com/layers/grafana/loki/master-364b5bc/images/sha256-4ad1606007c58baa1adff9c17ee2f698cf6e371774d79d8a0d2a1408b9d37df5 if you want to try it out.

**Comment by Horkyze** [comment]
Thanks, it is working now :) However with some small issue (which I resolved) https://github.com/grafana/grafana/issues/19764

**Comment by davkal** [comment]
The grafana team is working on this for 6.5. Sorry for the delay.

**Comment by cwrau** [comment]
This doesn't seem to work in 6.5, am I correct?

Can we maybe reopen this issue, if that is the case?

**Comment by cyriltovena** [comment]
It should work.

**Comment by cwrau** [comment]
I'm using Grafana 6.5.3 and Loki 1.2.0.

When I try to query Loki, I get the following error message: ```legacy endpoints only support streams result type, current type is matrix```

![image](https://user-images.githubusercontent.com/14326070/72727613-a9979d00-3b8b-11ea-84ed-c7f5d5a3a399.png)

Maybe I'm doing something wrong?

**Comment by cyriltovena** [comment]
Ho ! Does explore works ? It seems you’re using a graph panel, this works
only with Prometheus data source but Loki can be one if you add /loki at
the end of the data source path.

**Comment by cwrau** [comment]
Explore fails with the the same error:

![image](https://user-images.githubusercontent.com/14326070/72792689-b83c8d80-3c39-11ea-8ec9-0d2625fc208f.png)

When I add loki as a Prometheus it works, but I think that is suboptimal

**Comment by cyriltovena** [comment]
@ivanahuckova which Grafana version has native metrics support for Loki ?

@cwrau the thread above explain how to use loki as a prometheus data source in the meantime.

**Comment by cwrau** [comment]
I just tried 6.6.0-beta1 for #20300, and this works as well.

The only thing that's missing is the legend format that's available for Prometheus and alerting 😞

So even in 6.6.0-beta1 I'll need Loki as a Prometheus datasource, at least for alerting

**Comment by cyriltovena** [comment]
Yeah for alerting it will take time also, we use Loki as prom data source internally and it does the job.

**Comment by raffraffraff** [comment]
I've just installed from scratch on a new cluster using Helm. The missing legend format is a major issue and makes it Loki as a dashboard graphing tool useless to me, unless I have one series per panel, turn off legends and tooltips and use a sensible panel name. The default metric names are 300+ characters long, eg:

```{app="myappname", container="mycontainer", filename="/var/log/pods/default_myappname-cbbd8fc77-c6df7_852952d6-a6c6-4f24-91ff-dd4e4c76f5e0/myappname/0.log", job="namespace/myappname", namespace="namespace", pod="mypod-cbbd8fc77-c6df7", pod_template_hash="cbbd8fc77", stream="stderr"}
```

**Comment by cyriltovena** [comment]
The latest Grafana support this now.
